env_characterization/PathNode[] path